pkg/provider/resources/resources.go

```diff
@@ -11,6 +11,8 @@ const (
 	ApiAuthenticationIntegrationWithJwtBearer              resource = "snowflake_api_authentication_integration_with_jwt_bearer"
 	ApiIntegration                                         resource = "snowflake_api_integration"
 	AuthenticationPolicy                                   resource = "snowflake_authentication_policy"
+	Connection                                             resource = "snowflake_connection"
```

Comment:
I think it `snowflake_primary_connection` would be more clear. Same for file names, struct names, etc.